src/obb_rules/unit.cljc

```diff
@@ -22,10 +22,19 @@
             [obb-rules.units.star]))
 
 (defrecord ^{:doc "Represents a combat unit"} CombatUnit
-  [name code value
-   attack attack-type range defense
-   bonus type category displacement
-   movement-type movement-cost])
+  [^String name
+   ^String code
+   ^long value
+   ^long attack
+   attack-type
+   ^long range
+   ^long defense
+   bonus
+   type
+   category
+   displacement
+   movement-type
+   movement-cost])
```

Comment:
My bad!

A long is a primitive, and with such, the jvm can do primitive very fast math. With the Integer reference type, it has to box/unbox it. If it has no type hint, clojure will use heavy reflection to find out the type.

More details about performance issues: https://www.youtube.com/watch?v=iQwQXVM6oiY